include/seastar/core/sstring.hh

```diff
@@ -604,8 +604,11 @@ string_type uninitialized_string(size_t size) {
         return string_type(typename string_type::initialized_later(), size);
     } else {
         string_type ret;
-        // FIXME: use __resize_default_init if available
+#ifdef __cpp_lib_string_resize_and_overwrite
+        ret.resize_and_overwrite(size, [](string_type::value_type*, string_type::size_type) {});
+#else
```

Comment:
This is nice. We should implement basic_sstring::resize_and_overwrite() too, sstring should follow std::string wherever possible.

Reply:
added. but i think `sstring::resize_and_overwrite(pre_calculated_size, op)` is less performant than a combination of 
```c++
auto s = uninitialized_string(pre_calculated_size);
op(s.data(), s.size());
```

Reply:
I think the compiler will expand it to the same code.

Reply:
i doubt it. i think it depends on if the `sstring::resize_and_overwrite(pre_calculated_size, op)` is called on an instance of `sstring` with the right size. if we have something like
```c++
sstring s;
s.resize_and_overwrite(pre_calculated_size, op);
```
where `pre_calculated_size` is 16 which is greater than the default `max_size`. then we would have to reallocate the string.

Reply:
There's nothing to reallocate, since a default-constructed string is empty. It just allocates it once, just like initialized_later.

Reply:
ahh, true. the internal buffer is not used at all. so no need to reallocate the bits in it. sorry for missing this and thank you for explaining it to me.